### ISSUE TITLE
ver might be "null" for BES Flash detection

### ISSUE THREAD

**Issue by wchen-r7** [rapid7/metasploit-framework#5696]
In a BrowserExploitServer Flash exploit, if the requirement is written this way:

```ruby
:flash   => lambda do |ver|
  return true if Gem::Version.new(ver) <= Gem::Version.new('18.0.0.194')
end
```

You might hit this error:

"rex: Proc::on_request: ArgumentError: Malformed version number string null"

Backtrace example:

```
[07/10/2015 15:34:11] [e(0)] rex: Proc::on_request: ArgumentError: Malformed version number string null

/Users/sinn3r/.rvm/rubies/ruby-2.1.6/lib/ruby/site_ruby/2.1.0/rubygems/version.rb:206:in `initialize'
/Users/sinn3r/.rvm/rubies/ruby-2.1.6/lib/ruby/site_ruby/2.1.0/rubygems/version.rb:198:in `new'
/Users/sinn3r/.rvm/rubies/ruby-2.1.6/lib/ruby/site_ruby/2.1.0/rubygems/version.rb:198:in `new'
/Users/sinn3r/rapid7/msf/modules/exploits/multi/browser/adobe_flash_hacking_team_uaf.rb:73:in `block in initialize'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit/remote/browser_exploit_server.rb:248:in `call'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit/remote/browser_exploit_server.rb:248:in `block in get_bad_requirements'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit/remote/browser_exploit_server.rb:235:in `each'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit/remote/browser_exploit_server.rb:235:in `get_bad_requirements'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit/remote/browser_exploit_server.rb:506:in `on_request_uri'
/Users/sinn3r/rapid7/msf/lib/msf/core/exploit/http/server.rb:206:in `block in start_service'
/Users/sinn3r/rapid7/msf/lib/rex/proto/http/handler/proc.rb:38:in `call'
/Users/sinn3r/rapid7/msf/lib/rex/proto/http/handler/proc.rb:38:in `on_request'
/Users/sinn3r/rapid7/msf/lib/rex/proto/http/server.rb:364:in `dispatch_request'
/Users/sinn3r/rapid7/msf/lib/rex/proto/http/server.rb:298:in `on_client_data'
/Users/sinn3r/rapid7/msf/lib/rex/proto/http/server.rb:157:in `block in start'
/Users/sinn3r/rapid7/msf/lib/rex/io/stream_server.rb:48:in `call'
/Users/sinn3r/rapid7/msf/lib/rex/io/stream_server.rb:48:in `on_client_data'
/Users/sinn3r/rapid7/msf/lib/rex/io/stream_server.rb:192:in `block in monitor_clients'
/Users/sinn3r/rapid7/msf/lib/rex/io/stream_server.rb:190:in `each'
/Users/sinn3r/rapid7/msf/lib/rex/io/stream_server.rb:190:in `monitor_clients'
/Users/sinn3r/rapid7/msf/lib/rex/io/stream_server.rb:73:in `block in start'
/Users/sinn3r/rapid7/msf/lib/rex/thread_factory.rb:22:in `call'
/Users/sinn3r/rapid7/msf/lib/rex/thread_factory.rb:22:in `block in spawn'
/Users/sinn3r/rapid7/msf/lib/msf/core/thread_manager.rb:100:in `call'
/Users/sinn3r/rapid7/msf/lib/msf/core/thread_manager.rb:100:in `block in spawn'
```

This is because sometimes if the browser is Internet Explorer, has installed Flash, but has never enabled the ActiveX control, the version is still null. You might experience this during testing.

While having this problem, I tried the Adobe Flash's website and it found the version just fine. So I assume if the detection is done with a SWF, it should be fine. Only JavaScript-based detection (ours) suffers from this problem.

**Comment by jvazquez-r7** [comment]
Flash detection in AS here: https://github.com/wchen-r7/metasploit-framework/pull/32, hopefully it solves the issue!

**Comment by wchen-r7** [comment]
Resolved in #5650
